CHANGELOG.md

```diff
@@ -1,6 +1,20 @@
 CHANGELOG
 =========
 
+### 4.0.0 (2016-xx-xx)
+
+* Added: Interface for `Geocoder\Model\AddressCollection` called `Geocoder\GeocoderResult`. Public APIs are updated to type hint for `Geocoder\GeocoderResult`.
```

Comment:
I am not a fan of this to be honest. You repeat "geocoder" twice, and "result" does mean anything. I used `AddressCollection` exactly for that. It is meaningful and it is clear that we get a set of results. I don't see the point of changing this.